src/Main.elm

```diff
@@ -128,14 +147,41 @@ update msg model =
 
                         Nothing ->
                             NothingToSelect
+
+                ( m, cmd ) =
+                    update (FocusMovie selection) model
             in
-                ( { model
+                ( { m
                     | unwatched = remaining
                     , selectedMovie = selected
                   }
-                , Cmd.none
+                , cmd
                 )
```

Comment:
I find re-entering the update function confusing. I think this is clearer, and the duplication isn't too bad.
```elm
{ model 
  | focusedMovie = selected
  , unwatched = remaining
}, fetchMovie selected
```
Alternatively, for no duplication, I'd suggest creating:
```elm
openModal  modal movie =
      ({ model | focusedMovie = movie }, fetchMovie movie)
```
And when a movie is selected in update:
``` elm
openModal model movie
    |> Tuple.mapFirst (\model -> {model | unwatched = remaining})
```

Reply:
This is something I was struggling to get a message mapped to a command. I went down the route of trying to create a task, but that was adding a lot of complexity. I'll try this out.

Reply:
Oh, duh, since I just want to call another function I don't need to update or send another command, I just have the helper function. I was focusing too much on the message command idea.

Reply:
This seems to be a very common way for folks to get tripped up when starting to write apps in Elm. I've seen several threads about it online in various places.